nixos/modules/services/web-apps/overleaf.nix

```diff
@@ -0,0 +1,194 @@
+      FILESTORE_HOST = "localhost";
+      GRACEFUL_SHUTDOWN_DELAY = "0";
+      LISTEN_ADDRESS = "localhost";
+      MONGO_HOST = "localhost";
+      MONGO_CONNECTION_STRING = "mongodb://localhost:27017/overleaf";
+      NOTIFICATIONS_HOST = "localhost";
+      REALTIME_HOST = "localhost";
+      OVERLEAF_REDIS_PATH = "/run/redis-overleaf/redis.sock";
+      SPELLING_HOST = "localhost";
+      TRACK_CHANGES_HOST = "localhost";
+      WEBPACK_HOST = "localhost";
+      WEB_API_PASSWORD = "overleaf";
+      WEB_API_USER = "overleaf";
+      WEB_API_PORT = "3032";
+      WEB_API_HOST = "localhost";
+      WEB_HOST = "localhost";
+      WEB_PORT = "3032";
+      DATA_DIR = "/var/lib/overleaf";
+      NODE_ENV = "production";
+      OVERLEAF_CONFIG = "${pkgs.overleaf}/share/server-ce/config/settings.js";
+      OVERLEAF_FPH_DISPLAY_NEW_PROJECTS = "true";
+      OVERLEAF_MONGO_URL = "mongodb://localhost:27017/overleaf";
+      OVERLEAF_SESSION_SECRET = "hello";
+      OVERLEAF_HISTORY_BACKEND = "fs";
+      OVERLEAF_HISTORY_BLOBS_BUCKET = "/var/lib/overleaf/data/history/overleaf-blobs";
+      OVERLEAF_HISTORY_PROJECT_BLOBS_BUCKET = "/var/lib/overleaf/data/history/overleaf-project-blobs";
+      OVERLEAF_HISTORY_CHUNKS_BUCKET = "/var/lib/overleaf/data/history/overleaf-chunks";
+      OVERLEAF_HISTORY_ZIPS_BUCKET = "/var/lib/overleaf/data/history/overleaf-zips";
+      OVERLEAF_HISTORY_ANALYTICS_BUCKET = "/var/lib/overleaf/data/history/overleaf-analytics";
+      ADMIN_PRIVILEGE_AVAILABLE = "true";
+      OVERLEAF_SITE_URL = "https://verso.saumon.network";
+      OVERLEAF_ADMIN_EMAIL = "postmaster@mondon.me";
+      #OVERLEAF_SITE_LANGUAGE = "fr";
+      OVERLEAF_APP_NAME = "Verso, powered by SaumonNetwork";
+      #OVERLEAF_EMAIL_FROM_ADDRESS = "";
+      #OVERLEAF_EMAIL_SMTP_HOST = "";
+      #OVERLEAF_EMAIL_SMTP_PORT = "587";
+      #OVERLEAF_EMAIL_SMTP_SECURE = "true";
+      #OVERLEAF_EMAIL_SMTP_USER = "";
+      #OVERLEAF_EMAIL_SMTP_PASS = "";
+    };
+
+    systemd.services =
+      let activateServices = service: {
+        "overleaf-${service}" = {
+          description = "Overleaf ${service}";
+          wantedBy = [ "multi-user.target" ];
+          environment = cfg.settings;
+          path = with pkgs; [ texlive.combined.scheme-${cfg.texliveScheme} ];
+          serviceConfig = {
+            Type = "simple";
+            ExecStart = "${pkgs.overleaf}/bin/overleaf-${service}";
+            StateDirectory = "overleaf";
+            WorkingDirectory = "/var/lib/overleaf";
+            User = "overleaf";
+          };
+        };
+      };
+      in
+      mkMerge ((map activateServices [
+        "chat"
+        "clsi"
+        "contacts"
+        "docstore"
+        "document-updater"
+        "filestore"
+        "notifications"
+        "project-history"
+        "real-time"
+        "spelling"
+        "track-changes"
+        "web"
+      ]) ++
+      [{
+        ferretdb = mkIf (cfg.mongoDB == "ferretdb") {
```

Comment:
Early nitpick: I would probably split up the FerretDB specific stuff into another module in the following way:
```
config = mkMerge [{ general configuration ... } (mkIf cfg.mongoDB == "ferretdb" { FerretDB specific customizations... }]
```

this way, you would have less if-guards

Reply:
I'm doing a ferretDB module as we speak

Reply:
Amazing :) (but my comment still hold regarding minimizing the if-guards to make it easy to review FerretDB-specific customizations)